### PR TITLE
[bug][build] fix for make api_docs_build

### PR DESCRIPTION
**Pull request by lalanikarim**
#20972

Fixes issues with api docs build:
* exclude standard-tests from api_build_docs
* upgrade autodoc_pydantic to 1.9.0 to fix a migration related issue with BaseSettings

### CODE REVIEW

**Comment by lalanikarim** [comment]
@baskaryan, @efriis, @eyurtsev, @hwchase17 
comments appreciated on this PR. Does this fix an issue or is completely unnecessary?

**Comment by lalanikarim** [comment]
@baskaryan 
Can you please comment on whether this PR is useful and provide feedback?

File: docs/api_reference/requirements.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 -e libs/core
 -e libs/community
 pydantic<2
-autodoc_pydantic==1.8.0
+autodoc_pydantic==1.9.0
```
Comment (baskaryan) [inline docs/api_reference/requirements.txt:6]: whats the issue with 1.8.0?

Comment (lalanikarim) [reply docs/api_reference/requirements.txt:6]: I get this below error with 1.8.0 which doesn't happen with 1.9.x

```bash
API reference files built.
cd docs/api_reference && poetry run make html
Running Sphinx v4.5.0

Extension error:
Could not import extension sphinxcontrib.autodoc_pydantic (exception: `BaseSettings` has been moved to the `pydantic-settings` package. See https://docs.pydantic.dev/2.6/migration/#basesettings-has-moved-to-pydantic-settings for more details.

For further information visit https://errors.pydantic.dev/2.6/u/import-error)
make[1]: *** [html] Error 2
make: *** [api_docs_build] Error 2
```

**Comment by artemmukhin** [comment]
@baskaryan @efriis Could you please retake a look at this PR? 🙏 

I can confirm that, together with https://github.com/langchain-ai/langchain/pull/22973, it fixes the documentation build `$ make docs_build`. I can reproduce it on a fresh Codespaces environment from the main branch.

**Comment by lalanikarim** [comment]
This may no longer be needed. Closing.

**Pull request closed without merging**